.github/workflows/release.yml

```diff
@@ -0,0 +1,41 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "*"
+
+permissions:
+  contents: write
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    env:
+      CXX: clang++-14
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build
+        run: cmake -B build && cmake --build build
+      - name: Test
+        run: ctest --output-on-failure --test-dir build
+  release:
+    needs: build-and-test
+    runs-on: ubuntu-latest
+    if: ${{ needs.build-and-test.result == 'success' }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        
+    steps:
+      - uses: actions/checkout@v3
+      - name: Amalgamation
+        run: cd singleheader && ./amalgamate.py
+
+      - name: Create Release && Upload Assets
```

Comment:
I think we should just upload assets to a newly created release, since we might also want to add some description on our own before releasing it.

Reply:
Do you think it'd be worth automating the release notes creation, or is this something that is better being a manual thing?